File: src/NodeRTLib/DefinitonTemplates/Ts/Enum.ts
@{
@:export enum @(Model.Name) {
  var counter = 0;
  foreach(var field in Model.DeclaredFields)
  {
    if (counter != 0)
    {
    @:@(TX.Uncap(field.Name)),
    }
    counter++;
  }
  @:}
}

File: src/NodeRTLib/CppTemplates/Enum.cpp
  static void Init@(Model.Name)Enum(const Local<Object> exports)
  {
    HandleScope scope;
    
	Local<Object> enumObject = Nan::New<Object>();
    Nan::Set(exports, Nan::New<String>("@(Model.Name)").ToLocalChecked(), enumObject);
    @foreach(var name in Enum.GetNames(Model)) {
	@:Nan::Set(enumObject, Nan::New<String>("@(TX.Uncap(name))").ToLocalChecked(), Nan::New<Integer>(static_cast<int>(@TX.ToWinRT(Model)::@(name))));
    }
  }


